test/mod_websockets_SUITE.erl

```diff
@@ -153,3 +181,76 @@ get_ranch_connections() ->
     LSup = get_child_by_mod(ranch_sup, ranch_listener_sup),
     CSup = get_child_by_mod(LSup, ranch_conns_sup),
     [ {Mod, Pid} || {_, Pid, _, [Mod]} <- supervisor:which_children(CSup) ].
+
+wait_for_no_ranch_connections(Times) ->
+    case get_ranch_connections() of
+        [] ->
+            ok;
+        _ when Times > 0 ->
+            timer:sleep(100),
+            wait_for_no_ranch_connections(Times - 1);
+       Connections ->
+            error(#{reason => wait_for_no_ranch_connections_failed,
+                    connections => Connections})
+    end.
+
+%% ---------------------------------------------------------------------
+%% subprotocol_header_tests test functions
+%% ---------------------------------------------------------------------
+
+%% From RFC 6455:
+%%   The |Sec-WebSocket-Protocol| header field is used in the WebSocket
+%%   opening handshake.  It is sent from the client to the server and back
+%%   from the server to the client to confirm the subprotocol of the
+%%   connection.
+agree_to_xmpp_subprotocol(_) ->
+    check_subprotocol("Proper client behaviour", ["xmpp"], "xmpp").
+
+%% From RFC 6455:
+%%   The |Sec-WebSocket-Protocol| header field MAY appear multiple times
+%%   in an HTTP request (which is logically the same as a single
+%%   |Sec-WebSocket-Protocol| header field that contains all values).
+%%   However, the |Sec-WebSocket-Protocol| header field MUST NOT appear
+%%   more than once in an HTTP response.
+agree_to_xmpp_subprotocol_from_many(_) ->
+    check_subprotocol("Two protocols in one header", ["xmpp, other"], "xmpp"),
+    check_subprotocol("Two protocols in one header", ["other, xmpp"], "xmpp"),
+    check_subprotocol("Two protocols in two headers", ["other", "xmpp"], "xmpp").
```

Comment:
I don't think it's different form the one above. Did I miss sth?

Reply:
First is one header with two subprotocols (i.e. a list with one element)

Second is two headers with one protocol in each (two elements in the list)

Reply:
Now I see, thanks!